internal/bridge/crunchybridgecluster/crunchybridgecluster_controller_test.go

```diff
@@ -0,0 +1,156 @@
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package crunchybridgecluster
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"gotest.tools/v3/assert"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/crunchydata/postgres-operator/internal/testing/require"
+)
+
+func TestGetSecretKeys(t *testing.T) {
+	ctx := context.Background()
+	_, tClient := setupKubernetes(t)
+	require.ParallelCapacity(t, 0)
+
+	reconciler := &CrunchyBridgeClusterReconciler{Client: tClient}
+
+	ns := setupNamespace(t, tClient).Name
+	cluster := testCluster()
+	cluster.Namespace = ns
+
+	t.Run("NoSecret", func(t *testing.T) {
+		apiKey, team, err := reconciler.GetSecretKeys(ctx, cluster)
+		assert.Check(t, apiKey == "")
+		assert.Check(t, team == "")
+		assert.ErrorContains(t, err, "secrets \"crunchy-bridge-api-key\" not found")
+	})
+
+	t.Run("SecretMissingApiKey", func(t *testing.T) {
+		cluster.Spec.Secret = "secret-missing-api-key"
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "secret-missing-api-key",
+				Namespace: ns,
+			},
+			Data: map[string][]byte{
+				"team": []byte(`jkl;`),
+			},
+		}
+		assert.NilError(t, tClient.Create(ctx, secret.DeepCopy()))
+
+		apiKey, team, err := reconciler.GetSecretKeys(ctx, cluster)
+		assert.Check(t, apiKey == "")
+		assert.Check(t, team == "")
+		assert.ErrorContains(t, err, "error handling secret: found key false, found team true")
+
+		assert.NilError(t, tClient.Delete(ctx, secret.DeepCopy()))
+	})
+
+	t.Run("SecretMissingTeamId", func(t *testing.T) {
+		cluster.Spec.Secret = "secret-missing-team-id"
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "secret-missing-team-id",
+				Namespace: ns,
+			},
+			Data: map[string][]byte{
+				"key": []byte(`asdf`),
+			},
+		}
+		assert.NilError(t, tClient.Create(ctx, secret.DeepCopy()))
+
+		apiKey, team, err := reconciler.GetSecretKeys(ctx, cluster)
+		assert.Check(t, apiKey == "")
+		assert.Check(t, team == "")
+		assert.ErrorContains(t, err, "error handling secret: found key true, found team false")
+	})
+
+	t.Run("GoodSecret", func(t *testing.T) {
+		cluster.Spec.Secret = "crunchy-bridge-api-key"
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "crunchy-bridge-api-key",
+				Namespace: ns,
+			},
+			Data: map[string][]byte{
+				"key":  []byte(`asdf`),
+				"team": []byte(`jkl;`),
+			},
+		}
+		assert.NilError(t, tClient.Create(ctx, secret.DeepCopy()))
+
+		apiKey, team, err := reconciler.GetSecretKeys(ctx, cluster)
+		assert.Check(t, apiKey == "asdf")
+		assert.Check(t, team == "jkl;")
+		assert.NilError(t, err)
+	})
+
+}
+
+func TestDeleteControlled(t *testing.T) {
```

Comment:
I see a test for `Owned` in our other `TestDeleteControlled` test. Should we have one here :question: 

https://github.com/CrunchyData/postgres-operator/blob/master/internal/controller/postgrescluster/controller_test.go#L79-L90

Reply:
The test you are referring to tests the `setOwnerReference` function which we don't have/use in `crunchybridgecluster`.

Reply:
Then should the first test be `NotControlled` instead of `NoOwnership`?